lib/iris/tests/unit/plot/test_2d_coords.py

```diff
@@ -1,157 +0,0 @@
-# (C) British Crown Copyright 2018, Met Office
```

Comment:
I have removed this file as it had mixed unit tests for lots of different functions.
Unit tests should be separated so that each file tests a specific function/class (e.g. a single file for testing iris.plot._get_plot_defn)

I have moved the tests in this file to the appropriate files for *almost* all of them. Have not yet done it for the plot.py functions but will also push that up to this PR once that is done

Reply:
Cool, it was on my list of things to do.  In my own PR I have shuffled some tests around and added some more for the new utils, but these were always intended to be cleaned up anyway, it's just a bit easier to run tests when they're in the same place.  

I guess I can pull in your changes and then move the rest of the tests round accordingly, because none of them are in the right place.